File: var/www/cgi-bin/form-upload.rb
#!/usr/bin/ruby

require "cgi"

cgi = CGI.new
params = cgi.params

was_file_uploaded = false

if params.has_key?"file"
	file = params["file"].first

	if file && file.original_filename
		server_file = ENV['UPLOAD_FOLDER'] + file.original_filename

		if (!File.directory?(server_file))
			File.open(server_file.untaint, "w") do |f|
				f << file.read
			end

			was_file_uploaded = true
		end
	end
end

puts 'HTTP/1.0 200 OK'
puts 'Content-type: text/html'
puts ''

puts '<!DOCTYPE html>'
puts '<html lang="en">'
puts '	<head>'
puts '		<title>42 Webserver - Form Submission</title>'
puts '		<link rel="stylesheet" href="/index.css">'
puts '		<script src="/index.js"></script>'
puts '	</head>'
puts '	<body>'
puts '		<div class="page-wrapper">'
puts '			<div class="container">'
puts '				<section>'
puts '					<h1 class="title">'
puts "						File #{was_file_uploaded ? 'uploaded' : 'not uploaded'}"
puts '					</h1>'
puts '				</section>'
puts '			</div>'
puts '		</div>'
puts '	</body>'
puts '</html>'

File: var/www/cgi-bin/form-upload.py
#!/usr/bin/python

import cgi
import os

try:
	form = cgi.FieldStorage()

	was_file_uploaded = False

	if "file" in form and form["file"].file:
		file_item = form["file"]

		try:
			with open(os.environ.get('UPLOAD_FOLDER') + file_item.filename, "wb") as file:
				was_file_uploaded = True

				while True:
					chunk = file_item.file.read(1024)
					if not chunk:
						break
					file.write(chunk)
		except Exception as e:
			was_file_uploaded = False

	print('HTTP/1.0 200 OK')
	print('Content-type: text/html')
	print('')

	print('<!DOCTYPE html>')
	print('<html lang="en">')
	print('	<head>')
	print('		<title>42 Webserver - Form Submission</title>')
	print('		<link rel="stylesheet" href="/index.css">')
	print('		<script src="/index.js"></script>')
	print('	</head>')
	print('	<body>')
	print('		<div class="page-wrapper">')
	print('			<div class="container">')
	print('				<section>')
	print('					<h1 class="title">')
	print("						File {}".format('uploaded' if was_file_uploaded else 'not uploaded'))
	print('					</h1>')
	print('				</section>')
	print('			</div>')
	print('		</div>')
	print('	</body>')
	print('</html>')
except Exception as e:
	print("HTTP/1.0 500 Internal Server Error")